servers/user-api/src/resolvers/index.ts

```diff
@@ -91,6 +91,32 @@ export const resolvers = {
     description: (parent, args, context: IContext): Promise<string> => {
       return parent.description ?? context.models.user.description;
     },
+    premiumStatus: (
+      parent,
+      args,
+      context: IContext,
+    ): Promise<PremiumStatus> => {
+      return parent.premiumStatus ?? context.models.user.premiumStatus;
+    },
+    premiumFeatures: async (
+      parent,
+      args,
+      context: IContext,
+    ): Promise<PremiumFeature[] | null> => {
+      const status =
+        (await parent.premiumStatus) ??
+        (await context.models.user.premiumStatus);
+      if (status !== PremiumStatus.ACTIVE) {
+        return null;
```

Comment:
Im torn on null or empty array, but went null cause it should be an easier client check.